src/Language/PureScript/Sugar/Names.hs

```diff
@@ -244,14 +245,18 @@ renameInModule env imports (Module ss coms mn decls exps) =
     updateType :: Type -> m Type
     updateType (TypeOp name) = TypeOp <$> updateTypeOpName name pos
     updateType (TypeConstructor name) = TypeConstructor <$> updateTypeName name pos
-    updateType (ConstrainedType cs t) = ConstrainedType <$> updateConstraints pos cs <*> pure t
+    updateType (ConstrainedType cs t) = ConstrainedType <$> traverse updateInConstraint cs <*> pure t
     updateType t = return t
+    updateInConstraint :: Constraint -> m Constraint
+    updateInConstraint (Constraint name ts info) =
+      Constraint <$> updateClassName name pos <*> pure ts <*> pure info
```

Comment:
Like that above case, need a different version of `updateConstraints` here that does not recurse on `updateTypesEverywhere`, as the traversal already does that.

These two traversal hiccups became apparent as the naming pass is even less idempotent than it was before. I'm pretty sure a case could be constructed that would cause both of these to fail.